src/auto-complete.jsx

```diff
@@ -0,0 +1,296 @@
+        openDirection="bottom-left"
+        style={mergedMenuStyles}>
+        {
+          requestsList.map((request, index) => {
+            switch(typeof request){
+              case 'string':
+                return (<MenuItem
+                          disableFocusRipple={this.props.disableFocusRipple}
+                          innerDivStyle={{overflow:'hidden'}}
+                          key={index}
+                          value={request}
+                          primaryText={request}
+                          />);
+              case 'object':
+                return React.cloneElement(request, {
+                        key: index,
+                        disableFocusRipple: this.props.disableFocusRipple,
+                      });
+              default:
+                return null;
+            }
+          })
+        }
+      </Menu>
+    ) : null;
+
+    return (
+      <div style={mergedRootStyles}
+           onKeyDown={this._handleKeyDown}>
+        <div
+          style={{
+            width:'100%',
+          }}>
+          <TextField
+            {...other}
+            ref="searchTextField"
+            value={this.state.searchText}
+            onEnterKeyDown={()=>{
+                 setTimeout(() => {
+                   this.setState({open:false});
+                 }, this.props.touchTapCloseDelay);
+                this.props.onNewRequest(this.state.searchText);
+            }}
+            onChange={(e)=>{
+                let searchText = e.target.value;
+                this._updateRequests(searchText);
+            }}
+            onBlur={()=>{
+              if(this.focusOnInput && this.state.open)
+                this.refs.searchTextField.focus();
+            }}
+            onFocus={()=>{
+              if(!this.state.open && ( auto ||
+                  this.props.updateWhenFocused || this.state.searchText !== '')){
+                this._updateRequests(this.state.searchText);
+              }
+              this.focusOnInput = true;
+            }}
+
+            {...textFieldProps} />
+        </div>
+          <ReactTransitionGroup>{menu}</ReactTransitionGroup>
+      </div>
+    );
+  },
+
+  setValue(textValue) {
+    this.setState({
+      searchText: textValue,
+    });
+  },
+
+  getValue() {
+    return this.state.searchText;
+  },
+
+  _updateRequests(searchText){
+
+    this.setState({
+      searchText:searchText,
+    });
+    this.focusOnInput = true;
+
+    if(this.props.dataSource){
+      this.props.onUpdateInput(searchText);
+
+      let list = this.props.dataSource.filter((v) => v.includes(searchText));
+
+      if(this.props.auto && searchText === ''){
+        list = this.props.dataSource;
+      }
+      this.setState({
+        requestsList: list,
+        open: true,
+      })
+
+      return;
+    }
+
+
+    let requestsListPromise = new Promise((resolve, reject) => {
+      let list = this.props.onUpdateInput(searchText, this.state.requestsList);
```

Comment:
> My thought was if you want a shorter filter implementation with preloaded data, or work with simple request, this can be very handy to use

I understand your point. That's less boilerplate and more straightforward to use.
However, I think that having two ways to do the same thing is wrong.
First it's more code to maintain and second if you use the first more limited solution and at some point you need more power, you will have to refactor, not great.